src/components/calcite-link/calcite-link.scss

```diff
@@ -29,16 +28,15 @@
 }
 
 .calcite-link--icon {
-  transition: $transition;
+  @apply align-middle transition-default;
```

Comment:
Can you try the following? I think it will help set the icon size to inherit from its parent's font-size:
```scss
calcite-icon {
  width: 1em;
  height: 1em;
  min-width: unset;
  min-height: unset;
}

.calcite-link--icon {
  @apply transition-default align-text-top;
  margin-top: 0.125rem;
}
```